src/ui/resource/Image.js

```diff
@@ -80,12 +83,12 @@ exports = Class(lib.PubSub, function () {
 	var isNative = GLOBAL.NATIVE && !device.isNativeSimulator;
 	var Canvas = device.get('Canvas');
 
-	// helper canvases for filters and image data, initialized when/if needed
-	var _filterCanvas = null;
-	var _filterCtx = null;
+	// helper canvases for image data, initialized when/if needed
 	var _imgDataCanvas = null;
 	var _imgDataCtx = null;
 
+	this.filterRenderer = new FilterRenderer();
```

Comment:
Consider making `FilterRenderer` a singleton to better definite its purpose and use to other devs.